src/flow/darcy_flow_mh.cc

```diff
@@ -575,12 +583,32 @@ bool DarcyMH::zero_time_term(bool time_global) {
     }
 }
 
+double DarcyMH::compute_full_residual()
+{
+    Vec residual_;
+    VecDuplicate(schur0->get_solution(), &residual_);
+    MatMult(*schur0->get_matrix(), schur0->get_solution(), residual_);
+    VecAXPY(residual_,-1.0, *schur0->get_rhs());
+    double residual_norm;
+    VecNorm(residual_, NORM_2, &residual_norm);
+    VecDestroy(&residual_);
+    return residual_norm;
+}
+
+Vec DarcyMH::compute_full_residual_vec()
+{
+    Vec residual_;
```

Comment:
When residual_ is created inside this method, then it has to be deallocated manually on other place. It would be safer to create and deallocate the vector in the same scope and take residual_ as a parameter of the method compute_full_residual_vec.